test/Dashboard.EndToEndTests/Tests/ArgumentsDisplay/QueueArgumentsDisplayFunctions.cs

```diff
@@ -23,13 +23,13 @@ public class QueueArgumentsDisplayFunctions
         public static readonly MethodInfo SingletonMethodInfo = typeof(QueueArgumentsDisplayFunctions).GetMethod("Singleton");
 
         [Singleton("TestScope")]
-        public static void Singleton(
+        public static async Task Singleton(
             [QueueTrigger("queue-singleton-in")] string input,
-            [Queue("queue-singleton-out")] out string output,
-            [Queue(DoneNotificationFunction.DoneQueueName)] out string done)
+            [Queue("queue-singleton-out")] IAsyncCollector<string> output,
```

Comment:
You may be right for this one, I think the `Task.Run(() => RunTestHost(hostConfiguration)).GetAwaiter().GetResult();` change may resolve the Singleton-caused deadlock